### ISSUE TITLE
HTML does not define any TimeRange interface

### ISSUE THREAD

**Issue by nox** [w3c/media-source#195]
The specification mentions a `TimeRange` interface numerous times, pointing to [this W3C HTML interface](https://www.w3.org/TR/html51/semantics-embedded-content.html#timeranges-timeranges) ([WHATWG](https://html.spec.whatwg.org/multipage/#timeranges)), which is named `TimeRanges`, plural.

**Comment by chrisn** [comment]
It looks like the spec has been updated to use `TimeRanges` throughout, but the new language is confusing because in some cases the intent is to refer to a singular time range - e.g., in [3.15.3](https://w3c.github.io/media-source/#mediasource-seeking):

> If new playback position is not in any `TimeRanges` of `HTMLMediaElement`'s `buffered`

**Comment by joeyparrish** [comment]
I don't agree.  It refers to a single playback position, but multiple ranges.  `buffered` represents multiple disjoint time ranges, and the playback position could be within any of them.

**Comment by chrisn** [comment]
That's right, but "any TimeRanges" is what's confusing, as "any time range" (i.e., of the TimeRanges in buffered) would be more grammatical.

**Comment by joeyparrish** [comment]
Okay, I agree on grammar, then.

**Comment by jyavenard** [comment]
> because in some cases the intent is to refer to a singular time range

I don't think that's the intent. The intent is to state that currentTime is in none of the time intervals found in the buffered attribute.
